src/main/java/de/muenchen/isi/domain/service/calculation/SobonursaechlicheWohneinheitenService.java

```diff
@@ -0,0 +1,141 @@
+package de.muenchen.isi.domain.service.calculation;
+
+import de.muenchen.isi.domain.model.BaurateModel;
+import de.muenchen.isi.domain.model.calculation.WohneinheitenProFoerderartProJahrModel;
+import de.muenchen.isi.infrastructure.entity.enums.lookup.SobonOrientierungswertJahr;
+import de.muenchen.isi.infrastructure.repository.stammdaten.StaedtebaulicheOrientierungswertRepository;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class SobonursaechlicheWohneinheitenService extends WohneinheitenCalculationService {
+
+    private final FoerdermixUmlageService foerdermixUmlageService;
+
+    private final StaedtebaulicheOrientierungswertRepository staedtebaulicheOrientierungswertRepository;
+
+    /**
+     * Errechnet SoBoN-ursächliche Wohneinheiten je Förderart und Jahr.
+     * Ursprungsinfo ist die SoBoN-ursächliche Geschossfläche in der Abfragevariante
+     * Die Fördermixe aller Bauraten werden vor der Berechnung umgelegt, siehe {@link FoerdermixUmlageService}.
+     *
+     * Ist das Ergebnis aus dieser Rechnung über 1000 Wohneinheiten, werden die Wohneinheiten auf mehrere Bauraten (mehrere Jahre) aufgeteilt.
+     * Es wird immer bei 1000 WE ein Schnitt gemacht.
+     *
+     * Für die Summenbildung gibt es eine zusätzliche "Förderart", die die Summen aller Förderarten pro Jahr enthält.
+     * Außerdem gibt es zusätzliche Jahre pro Förderart, die über 10, 15 & 20 Jahre die Wohneinheiten aufsummieren.
+     *
+     *
+     * @param sobonGF SoBoN-ursächliche Geschossfläche in der Abfragevariante
+     * @param baurate Baurate (erstes Jahr). Werden mehrere Bauabschnitte angelegt, wird vom ersten Bauabschnitt,
+     *         der angelegt wurde, das erste Baugebiet, das angelegt wurde und dann davon die erste Baurate verwendet.
+     * @param sobonJahr Das SoBoN-Jahr, welches die städtebaulichen Orientierungswerte diktiert.
+     * @param gueltigAb Das Gültigkeitsdatum der Stammdaten, welche die Umlegung diktieren.
+     * @return Eine Liste von {@link WohneinheitenProFoerderartProJahrModel}, welche alle Wohneinheiten pro Förderart und Jahr darstellt.
+     */
+    public List<WohneinheitenProFoerderartProJahrModel> calculateSobonursaechlicheWohneinheiten(
+        final BigDecimal sobonGF,
```

Comment:
Klassen- sowie Methodenvariablen, Methodenparameter, Methoden, Klassen immer in CamelCase-Schreibweise benennen.

`sobonGF` -> `sobonGf`